lib/ui/painting/fragment_program.cc

```diff
@@ -69,6 +69,7 @@ fml::RefPtr<FragmentShader> FragmentProgram::shader(
   for (size_t i = 0; i < uniform_count; i++) {
     uniform_floats[i] = uniforms[i];
   }
+  uniforms.Release();
```

Comment:
We need to do an explicit release before we call any other Dart API. But if we don't call any Dart API, it's fine to not explicitly release the data.

IMHO, the tonic typed data class(es) are poorly designed, mainly because the API they are wrapping is difficult to use correctly. One thing we could try to do is avoid having automatic conversion of these in tonic, since misuse tends to happen where it's unclear that you have to free the passed in argument before you do any other dart_api calls that may allocate anything.

Reply:
I've done a couple experiments and filed https://github.com/flutter/flutter/issues/99431. I think it's worth refactoring this further but it should probably be done incrementally and is a bigger task than I can take on at the moment.